115 webpack integration

anthonythorne opened the pull request:
This webpack integration is similar yet different from our usual.

Due to the nature of this being a plugin, I have the build tools dir external to the plugin, along with the asset src code.

The compiled code is automatically cleaned and placed inside of the plugin dir.

This approach is so that when the plugin dir is the source of download/update our source files and build process is not included. Just the compiled asset files.

README has been updated accordingly, along with detailed instructions of build and dev processes.

ENQUEUS:
The enqueued is done in a manner that it auto-detects the local environment and file extension that exists based of the main site's config variables and attempting to load the standard file before minified if local. otherwise always loading the compressed file. This is good if someone pulls the repo to make changes, that are not asset-related, they can have the latest asset files being loaded without running dev build or new prod build.

Notes; other changes from our normal approach.
Uglify JS would not work with ES6, brought up yesterday with front end devs slack channel. I have changed it for Terser webpack plugin.
I have removed browser sync as this is a plugin not a site and the domain will always be different.
I have removed all image related compression etc, this can be added back later when images are included.

TESTED:
Tested the admin build scripts are working.

anthonythorne committed:
Add the build tools scripts and files

anthonythorne committed:
Build tools working with upgrade for uglify JS to terser

anthonythorne committed:
Webpack integration for build process. Tested and working

anthonythorne committed:
Add new lines at the end of files.

BronsonQuick reviewed (approved):
Nicely implemented! I did see some of this functionality in your framework so thanks for porting it over. I'll leave this open so Zac can review it as well.

anthonythorne committed:
remove lock files, add to git ignore. update script file to include composer in plugin again.

anthonythorne committed:
Merge branch 'master' into 115-webpack-integration

anthonythorne committed:
Add hidden fields for the product and components

anthonythorne committed:
Revert "Add hidden fields for the product and components"

This reverts commit 19f361fb48ed138e9ee552ca58433cf21944ea93.

anthonythorne committed:
Revert "Add note for the manipulation of data for products."

This reverts commit e7baf53c9a08b8493fcff64b5fb269a658c3351b.

anthonythorne committed:
Add note for the manipulation of data for products.

anthonythorne commented:
@zacscott are you happy for this to be included, there has been a couple of PRs that has been based off this branch.
